indra/newview/llstartup.cpp

```diff
@@ -938,17 +938,28 @@ bool idle_startup()
 		LLFile::mkdir(gDirUtilp->getLindenUserDir());
 
 		// As soon as directories are ready initialize notification storages
-		if (!LLPersistentNotificationStorage::instanceExists())
+		if (LLPersistentNotificationStorage::isInstanceUninitialised())
 		{
 			// check existance since this part of code can be reached
 			// twice due to login failures
 			LLPersistentNotificationStorage::initParamSingleton();
-			LLDoNotDisturbNotificationStorage::initParamSingleton();
```

Comment:
I'm sorry, this feels to me like flailing.

Can we zoom out a notch? Why are `LLPersistentNotificationStorage` and `LLDoNotDisturbNotificationStorage` derived from `LLParamSingleton` instead of `LLSingleton`, since neither constructor requires any arguments? Why call `initParamSingleton()` here at all, under various combinations of circumstances? Why not just let any subsequent `instance()` or `getInstance()` call initialize these singleton classes?

Reply:
Originally it was a static class, then a param singleton to have a strict initialization time, so that people won't attempt to access it too early. It probably can be safely converted to a LLSingleton.